targets/TARGET_Cypress/TARGET_PSOC6/pwmout_api.c

```diff
@@ -156,59 +156,78 @@ void pwmout_init(pwmout_t *obj, PinName pin)
 
     MBED_ASSERT(obj);
     MBED_ASSERT(pin != (PinName)NC);
-    // Allocate and setup clock.
+    
+    // Allocate and setup clock (same clock for all PWMs)
     if (pwm_clock_divider == CY_INVALID_DIVIDER) {
+        
         pwm_clock_divider = cy_clk_allocate_divider(CY_SYSCLK_DIV_8_BIT);
         if (pwm_clock_divider == CY_INVALID_DIVIDER) {
             error("PWM clock divider allocation failed.");
-            return;
         }
-        Cy_SysClk_PeriphSetDivider(CY_SYSCLK_DIV_8_BIT,
-                                   pwm_clock_divider,
-                                   (CY_CLK_PERICLK_FREQ_HZ / PWMOUT_BASE_CLOCK_HZ) - 1);
+        
+        /* Configure divider */
+        Cy_SysClk_PeriphSetDivider(CY_SYSCLK_DIV_8_BIT, pwm_clock_divider,
+                                   (cy_PeriClkFreqHz / PWMOUT_BASE_CLOCK_HZ) - 1);
         Cy_SysClk_PeriphEnableDivider(CY_SYSCLK_DIV_8_BIT, pwm_clock_divider);
     }
 
+    /* Find instance using pins */
     pwm_cnt = pinmap_peripheral(pin, PinMap_PWM_OUT);
     if (pwm_cnt != (uint32_t)NC) {
-        if (cy_reserve_io_pin(pin)) {
-            error("PWMOUT pin reservation conflict.");
-        }
+        
+        obj->pin  = pin;
         obj->base = (TCPWM_Type*)CY_PERIPHERAL_BASE(pwm_cnt);
-        obj->pin = pin;
         if (obj->base == TCPWM0) {
+            /* TCPWM0 is used */
             obj->counter_id = ((PWMName)pwm_cnt - PWM_32b_0) / (PWM_32b_1 - PWM_32b_0);
             abs_cnt_num = obj->counter_id;
         } else {
-            // TCPWM1 is used.
+            /* TCPWM1 is used */
             obj->counter_id = ((PWMName)pwm_cnt - PWM_16b_0) / (PWM_16b_1 - PWM_16b_0);
             abs_cnt_num = obj->counter_id + 8;
         }
-        if (cy_reserve_tcpwm(abs_cnt_num)) {
-            error("PWMOUT Timer/Counter reservation conflict.");
+        
+        /* Check if resource severed */
+        if (0 != cy_reserve_tcpwm(abs_cnt_num))
+        {
+            Cy_TCPWM_PWM_Disable(obj->base, obj->counter_id);
+            Cy_TCPWM_PWM_DeInit (obj->base, obj->counter_id, &pwm_config);
+        }
+        else
```

Comment:
Not requirement at the moment, but I can see `{` starts on the new line here , not on thel ine 198 or others. If you can run astyle on HAL implementation, would be good to align the code

Reply:
If run with astyle, it should address also above review about tabs/space alignment

Reply:
Applied astyle to the HAL implementation, now the style is aligned with mbed-os: https://github.com/ARMmbed/mbed-os/pull/9481/commits/e4e5f6538e8e2771de4dcf2f3e6255b14ede043c.
Note: PDL is using different code style (to match the company guidelines), no plan to reformat peripheral driver sources and device headers.
I plan to apply astyle to feature/mbedtls and feature/FEATURE_BLE targets on consecutive pull requests.

Reply:
> Note: PDL is using different code style (to match the company guidelines), no plan to reformat peripheral driver sources and device headers.

Yes, please don't. Use astyleignore (currently ignored all in targets). We would like to allow HAL astyle checking in the future (no 3rd party drivers like PDL)

Reply:
Understood. Will notify mbedtls and BLE developers to apply mbed astyle to new Cypress code outside of /targets, and add astyle hook to internal CI to detect issues in our HAL implementation.